lib/connect.c

```diff
@@ -660,7 +662,15 @@ UNITTEST bool getaddressinfo(struct sockaddr *sa, char *addr,
 #if defined(HAVE_SYS_UN_H) && defined(AF_UNIX)
     case AF_UNIX:
       su = (struct sockaddr_un*)sa;
-      msnprintf(addr, MAX_IPADR_LEN, "%s", su->sun_path);
+      /* On Linux, sun_path within struct sockaddr_un is not initialized
+         if an unbound AF_UNIX socket is returned from accept() or
+         getsockname()/getpeername() are invoked for an unbound socket.
+         Evaluate the address length to not access sun_path in this case. */
+      if(socklen > offsetof(struct sockaddr_un, sun_path))
+        msnprintf(addr, MAX_IPADR_LEN, "%s", su->sun_path);
```

Comment:
This check puzzles me. You say `sun_path` ends up uninitialized here and that's the situation the condition here tries to avoid.

But how is checking if the storage struct is bigger than the offset to the `sun_path` field in the `sockaddr_un` a good check for this? This seems to rely on the representation of a struct that could vary and a destination buffer size that also could vary, to detect something that is unrelated to those two values.

Reply:
The variable is actually misnamed here, only seeing that now. `addrlen` oder `salen` would be better here. `accept()` and `getpeername()`/`getsockname()` return (via a pointer) the length of the address, i.e. the amount of initialized data, which is usually equal to the size of the corresponding socket adress struct for that address family, but not so for AF_LOCAL, where the struct is partially initialized only, due to the variable length `sun_path`.

On my linux system, getsockname() would return 2 for an unbound AF_LOCAL socket.

Taking the definition of `struct sockaddr_un`:
```
struct sockaddr_un {
        __kernel_sa_family_t sun_family; /* AF_UNIX */
        char sun_path[UNIX_PATH_MAX];   /* pathname */
};
```
Hence the address length of 2 indicates that only `sun_family` but not `sun_path` are initialized.